src/main.rs

```diff
@@ -92,8 +92,15 @@ fn main() {
 	
 	// Below unwrap()'s are safe within Docopt's usage rules
 
+	// Create data access object for data retrieval from database, file, etc.
+	// Use only Postgres for now
+	let psql_dao = match DaoPostgres::new() {
+		Ok(d) => d,
+		Err(e) => panic!("Failed to retrieve DAO: {}", e)
+	};
+
 	// Every proton command is mapped to a specific function that should be run
-	let command: fn(Args) -> Result<ProtonReturn, Error> = match env::args().nth(1).unwrap().as_ref() {
+	let command: fn(Args, DaoPostgres) -> Result<ProtonReturn, Error> = match env::args().nth(1).unwrap().as_ref() {
```

Comment:
Do all of these functions need access to the database? I think they do, but just double checking.

Reply:
Good check. Confirmed that all need the DAO